subsys/bluetooth/host/smp.c

```diff
@@ -5416,6 +5421,16 @@ int bt_conn_set_bondable(struct bt_conn *conn, bool enable)
 {
 	struct bt_smp *smp;
 
+	if (IS_ENABLED(CONFIG_BT_CLASSIC) && (conn->type == BT_CONN_TYPE_BR)) {
```

Comment:
```suggestion
	if (IS_ENABLED(CONFIG_BT_CLASSIC) && IS_ENABLED(CONFIG_BT_BONDABLE) && (conn->type == BT_CONN_TYPE_BR)) {
```

Reply:
For the implementation and comments of function `bt_conn_set_bondable`, I think the `CONFIG_BT_BONDABLE` only affects the initialisation value.

Reply:
`CONFIG_BT_BONDABLE` enables the entire bonding feature in LE:

```
#if defined(CONFIG_BT_BONDABLE)
#define BT_SMP_AUTH_BONDING_FLAGS BT_SMP_AUTH_BONDING
#else
#define BT_SMP_AUTH_BONDING_FLAGS 0
#endif /* CONFIG_BT_BONDABLE */
```

I think there are probably some functions (like `bt_set_bondable`) that should be guarded with this, as if `CONFIG_BT_BONDABLE=n` then the `bondable` value is effectively not used. 

In that regard it does not make sense to perform any of the checks here if it's disabled.

Reply:
I think it is not at all. When `CONFIG_BT_BONDABLE=n` and `bt_conn_set_bondable(conn, true)` called, the `auth_req` of pairing request or security request will set bit `BT_SMP_AUTH_BONDING`.
Please refer to https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/smp.c#L2889 or https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/smp.c#L3135.

https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/smp.c#L2670-L2674

Reply:
@jhedberg You are also more than welcome to weigh in here. Does it looks like there's an issue with `BT_BONDABLE` if it can be set to 0 but it's still possible to be bondable via `bt_conn_set_bondable`?

Reply:
@Thalley the way I see it, is that the Kconfig option is about supporting bondable mode, whereas the runtime API is about enabling/disabling it.

Reply:
> @Thalley the way I see it, is that the Kconfig option is about supporting bondable mode, whereas the runtime API is about enabling/disabling it.

If we currently allow toggling bondable even with the Kconfig disabled, then this interpretation doesn't seem correct (and the Kconfig text is misleading)

Reply:
We have this:
```c
static bool bondable = IS_ENABLED(CONFIG_BT_BONDABLE);
```

So the meaning of the current Kconfig option is just about setting the default state of `bondable`. This is not reflected in the Kconfig help text.

Another issue I see in `host/smp.c` is stuff like this, which seems to completely ignore any runtime state:
```c
#if defined(CONFIG_BT_BONDABLE)
#define BT_SMP_AUTH_BONDING_FLAGS BT_SMP_AUTH_BONDING
#else
#define BT_SMP_AUTH_BONDING_FLAGS 0
#endif /* CONFIG_BT_BONDABLE */
```

Reply:
Right, so the implementation does not match the description of the Kconfig option. So we have to adjust either of them

Reply:
After reading through the `smp.c` code a bit more it seems like the Kconfig help text should probably be updated. The bonding flags macros also don't seem to have much effect, rather the code will overwrite those both ways depending on the runtime setting. Anyway, all of that seems to be out of scope for this PR, so I'll approve here.

Reply:
Indeed, it is outside the scope of this PR, the question is mostly whether this PR should consider `CONFIG_BT_BONDABLE` per the text or per the implementation.

Reply:
> Indeed, it is outside the scope of this PR, the question is mostly whether this PR should consider `CONFIG_BT_BONDABLE` per the text or per the implementation.

Sorry, I missed the detail where this actually has different behavior than the existing implementation, i.e. uses Kconfig to guard against the entry into this BR/EDR branch. @lylezhu2012 could you fix that?

Reply:
> @lylezhu2012 could you fix that?

FWIW, feel free to add a second commit here to fix the Kconfig help text as well, if you want (otherwise it'll need to be fixed in a follow-up PR).

Reply:
Yes. I have added a commit to update it.